src/QsCompiler/Transformations/LiftLambdas.cs

```diff
@@ -116,97 +117,58 @@ public override TypedExpression OnTypedExpression(TypedExpression ex)
                     }
                 }
 
-                private ParameterTuple MakeLambdaParams(ResolvedType expressionType, QsSymbol paramNames)
+                private ParameterTuple MakeLambdaParams(ParameterTuple paramNames)
```

Comment:
The original purpose of this function was convert from the untyped parameters to a typed ParameterTuple. Since that is redundant now, the function's purpose has changed a bit. I suggest to rename this something like `SanitizeLambdaParams` or something similar. The nested functions can be named very similarly as well, maybe just `_sanitizeLambdaParams` or `SanitizeLambdaParamsRec`.

Reply:
Renamed.